src/Core/Event.js

```diff
@@ -14,32 +14,42 @@ var Event = {
 
 		this[type].set(DOMNode(element), element)
 	},
+	dispatchEvent: function (element, type, event) {
+		try {
+			var host = element.host
+			var instance = host.instance
+			var callback = element.event[type]
+			var state = null
+
+			if (!callback)
+				return
+
+			if (typeof callback === 'function')
+				state = callback.call(instance, event)
+			else if (typeof callback.handleEvent === 'function')
+				state = callback.handleEvent(event)
+
+			if (instance && state)
+				lifecycleReturn(host, state)
+		} catch (e) {
+			errorBoundary(host, e, type+':'+getDisplayName(callback.handleEvent || callback), 0)
+		}
+	},
 	/**
 	 * @param {Event} event
 	 */
 	handleEvent: function handleEvent (event) {
 		var type = 'on'+event.type
-		var element = this[type].get(event.target)
-
-		if (element)
-			try {
-				var host = element.host
-				var instance = host.instance
-				var callback = element.event[type]
-				var state = null
-
-				if (!callback)
-					return
-
-				if (typeof callback === 'function')
-					state = callback.call(instance, event)
-				else if (typeof callback.handleEvent === 'function')
-					state = callback.handleEvent(event)
+		var target = event.target
```

Comment:
Recommend using Reacts method of event target extraction
https://github.com/facebook/react/blob/ac9d6986364fd94cd8dd0dde219b2a32a17d60e2/src/renderers/dom/shared/utils/getEventTarget.js#L23